### PR TITLE
m32 - fspoly index and rounding - do not merge

### PR DESCRIPTION
**Pull request by feilipu**
This PR will be replaced by a smaller one that maintains the current calling convention for `poly()` as defined in the IEEE math standard.

- [x] Reworked the polynomial evaluation function to find the highest index for the table in the first byte of the table. With the index calculation, a maximum of 64 coefficients are supported. The maximum table index used in the library is currently 9, so this won't be a limitation.
- [x] Reworked the rounding to match rest of the library.

And, you know, the output is pretty good, at least for logf and sinf. It agrees with the d48 calculations very well, except for the expf function that is very fast moving. I don't think that reworking coefficients is a priority, now that the underlying math is solid.

There may be breakage for sccz80. I've tried create the right cm32 stubs, but this needs checked.

### CODE REVIEW

**Comment by suborb** [comment]
I think making the internal functions callee will have broken the bindings for both compilers - I removed the modifier in my PR to simplify things.

When a function is callee we have to do this trick (random function chosen):

```
extern char __LIB__ *utoa(uint16_t num,char *buf,int radix) __smallc;
extern char __LIB__ *utoa_callee(uint16_t num,char *buf,int radix) __smallc __z88dk_callee;
#define utoa(a,b,c) utoa_callee(a,b,c)
```

So that when we call directly we call the callee function, but when we take a pointer we take the address of the non-callee version. This means we need a cm32_XXXX_YYYY_callee.asm stub.

There's a similar thing that has to be done for __z88dk_fastcall as well.

However, it will be currently working since the .asm files haven't been regenerated - I left this as a manual step so there's no hard dependency on sdcc when building - just run the make in math32/c and commit the changes.

I also spotted that m32_atanf() is actually named m32_m32_atanf() which is probably due to my bad sedding.

**Comment by feilipu** [comment]
Ah. OK. I didn't notice your active change of the linkage.  I just thought it was my sloppiness.
I'll unwind that out of this PR. Sorry.

Also, I'll leave it open so that it doesn't mess up any current work. It can be committed just when someone is working on the #1155.

File: libsrc/_DEVELOPMENT/math/float/math32/c/asm/m32_tanf.asm [hunk 1]
```diff
@@ -274,15 +278,25 @@ _m32_tanf:
 	add	ix,sp
 	push	af
 	push	af
+	ld	a, l
+	ld	(), a
```
Comment (feilipu) [inline libsrc/_DEVELOPMENT/math/float/math32/c/asm/m32_tanf.asm]: Breakage.

File: libsrc/_DEVELOPMENT/math/float/math32/c/asm/m32_tanf.asm [hunk 2]
```diff
@@ -274,15 +278,25 @@ _m32_tanf:
...
+	ld	a, h
+	ld	(), a
```
Comment (feilipu) [inline libsrc/_DEVELOPMENT/math/float/math32/c/asm/m32_tanf.asm]: Breakage.

**Comment by feilipu** [comment]
Also noting that there is a significant speed up and accuracy improvement in replacing the many`1.0/x` cases in the c code by `m32_invf()`. And the `x*x` cases with `m32_sqrf()’.

A separate PR I think.

**Comment by feilipu** [comment]
Closed because IEEE set  the calling convention for `poly()` and there's no reason to reinvent that wheel.

**Pull request closed without merging**